docs/docs-components/DocsSideNavigation.js

```diff
@@ -18,40 +18,78 @@ export function isComponentsActiveSection(pathname: string): boolean {
 }
 
-export default function DocsSideNavigation({ border }: {| border?: boolean |}): Node {
+export default function DocsSideNavigation(): Node {
   const [activeSection, setActiveSection] = useState(newSidebarIndex[0]);
 
-  const router = useRouter();
-  const { componentPlatformFilteredBy, setComponentPlatformFilteredByCookie, setIsSidebarOpen } =
-    useNavigationContext();
+  const { isMobile } = useDocsDeviceType();
+  const { pathname } = useRouter();
+  const {
+    componentPlatformFilteredBy,
+    setComponentPlatformFilteredByCookie,
+    setIsSidebarOpen,
+    selectedTab,
+    setSelectedTab,
+  } = useNavigationContext();
 
   // Find the section that corresponds to the top navigation
   // If it's the components section, find the section that is currently
   // filtered with the component platform switcher
+
+  const isComponentsSection = isMobile
+    ? selectedTab === 'Components'
+    : isComponentsActiveSection(pathname);
+
+  const platformSwitcher = isComponentsSection ? (
+    <SidebarPlatformSwitcher
+      componentPlatformFilteredBy={componentPlatformFilteredBy}
+      onClick={(platform) => setComponentPlatformFilteredByCookie(platform)}
+    />
+  ) : null;
+
   useEffect(() => {
-    const sectionToRender = isComponentsActiveSection(router.pathname)
-      ? newSidebarIndex.find((section) =>
-          convertNamesForURL(section.sectionName).includes(componentPlatformFilteredBy),
-        )
-      : newSidebarIndex.find((section) =>
-          router.pathname.includes(convertNamesForURL(section.sectionName)),
-        );
-    setActiveSection(sectionToRender || newSidebarIndex[0]);
-  }, [router.pathname, componentPlatformFilteredBy]);
+    const isComponentsCallback = (section) =>
+      convertNamesForURL(section.sectionName).includes(componentPlatformFilteredBy);
+
+    const isNotComponentsCallback = (section) =>
+      isMobile
+        ? section.sectionName === selectedTab
+        : pathname.includes(convertNamesForURL(section.sectionName));
+
+    const sectionToRender =
+      newSidebarIndex.find(isComponentsSection ? isComponentsCallback : isNotComponentsCallback) ??
+      newSidebarIndex[0];
+
+    setActiveSection(sectionToRender);
+  }, [isMobile, isComponentsSection, pathname, componentPlatformFilteredBy, selectedTab]);
+
+  const sectionItemsForSideNav = useGetSideNavItems({ sectionInfo: activeSection });
 
-  const sectionItemsForSideNav = useGetSideNavItems((activeSection: siteIndexType));
   const closeSideNavigation = () => setIsSidebarOpen(false);
 
   return (
     <SideNavigation
       accessibilityLabel="Page navigation"
-      showBorder={border}
       header={
-        isComponentsActiveSection(router.pathname) && (
-          <SidebarPlatformSwitcher
-            componentPlatformFilteredBy={componentPlatformFilteredBy}
-            onClick={(platform) => setComponentPlatformFilteredByCookie(platform)}
-          />
+        isMobile ? (
+          <Flex direction="column" gap={4}>
```

Comment:
might be helpful to pull this out the same way you pulled out the platform switcher to make this bit easier to read